### PR TITLE
Add run_wine_command for better handling of wine

### PR DESCRIPTION
**Pull request by wannkunstbeikor**
This probably sucks ass, since its my first time using rust, but it seems to work

### CODE REVIEW

File: maxima-lib/src/unix/wine.rs [hunk 1]
```diff
@@ -82,6 +81,54 @@ fn get_wine_release() -> Result<GithubRelease> {
...
+        .env("WINEPREFIX", wine_prefix_dir()?)
+        .arg("--wait");
+    wine_server.spawn()?.wait()?;
+
+    if status.success() {
+        return Ok(());
+    }
+
+    let code = status.code();
+
```
Comment (battledash) [inline maxima-lib/src/unix/wine.rs]: Remove newline

File: maxima-lib/src/unix/wine.rs [hunk 2]
```diff
@@ -215,22 +278,8 @@ fn extract_dynamic_archive<R>(reader: R, label: &str, version: &str) -> Result<(
...
-        .arg("C:/wine.reg")
-        .output()
-        .expect("Failed to execute wine while modifying the registry");
-
-    let output_str = String::from_utf8_lossy(&output.stderr);
-    if !output.status.success() && !output_str.is_empty() {
-        bail!("Failed to modify the wine registry: {}", output_str);
-    }
+    run_wine_command("wine", "reg", Some(vec!["add", "HKLM\\Software\\Electronic Arts\\EA Desktop", "/v", "InstallSuccessful",  "/d", "true", "/f", "/reg:64"]))?;
+    run_wine_command("wine", "reg", Some(vec!["add", "HKLM\\Software\\Origin", "/v", "ClientPath",  "/d", "C:/Windows/System32/conhost.exe", "/f", "/reg:32"]))?;
```
Comment (battledash) [inline maxima-lib/src/unix/wine.rs:280]: Are you sure the reg:32 and reg:64 don't need to be swapped here? Afaik in 64-bit mode entries are placed into Wow6432Node, and that needs to happen with the Origin key, not the EA Desktop one. Can you verify with regedit in the wineprefix if the Origin key is placed in Wow6432Node here?

**Review by battledash — CHANGES_REQUESTED** [review]
Looks good, just a couple notes! Also, please capitalize the comments.

File: maxima-lib/src/unix/wine.rs (hunk 2 above)
Comment (wannkunstbeikor) [reply maxima-lib/src/unix/wine.rs:280]: yeah i checked, it works like this

Comment (wannkunstbeikor) [reply maxima-lib/src/unix/wine.rs:280]: > When a 32-bit application queries a value under the HKEY_LOCAL_MACHINE\SOFTWARE\<company>\<product> subkey, the application reads from the HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\<company>\<product> subkey.

Comment (wannkunstbeikor) [reply maxima-lib/src/unix/wine.rs:280]: from https://www.itprotoday.com/windows-78/whats-wow6432node-under-hkeylocalmachinesoftware-registry-subkey

Comment (battledash) [reply maxima-lib/src/unix/wine.rs:280]: Ah kk

File: maxima-lib/src/unix/wine.rs (hunk 1 above)
Comment (wannkunstbeikor) [reply maxima-lib/src/unix/wine.rs]: which one, 128?

Comment (wannkunstbeikor) [reply maxima-lib/src/unix/wine.rs]: or should i just do 
`bail!("{}", status.code().unwrap());`

Comment (battledash) [reply maxima-lib/src/unix/wine.rs]: Yeah that works too

**Review by battledash — APPROVED** [review]

**Pull request merged**